src/IECoreImage/OpenImageIOAlgo.cpp

```diff
@@ -438,7 +438,15 @@ DataView::DataView( const IECore::Data *d, bool createUStrings )
 			);
 			data = static_cast<const M44fVectorData *>( d )->baseReadable();
 			break;
-
+		case StringVectorDataTypeId:
+			type = TypeDesc(
+				TypeDesc::STRING,
+				TypeDesc::SCALAR,
+				TypeDesc::NOSEMANTICS,
+				static_cast<const StringVectorData *>( d )->readable().size()
+			);
+			data = static_cast<const StringVectorData *>( d )->baseReadable();
```

Comment:
I presume this requires #690 ?

Reply:
> I presume this requires #690 ? 

Yup sorry I should have mentioned.

Reply:
I don't think this is right. OIIO expects to receive a pointer to a `char *` for `TypeDesc::STRING`, but here we're passing a pointer to a `std::string`. It just so happens that with libstdc++ that the layout of a `std::string` is compatible with that of a `char *` so you can get away with it on Linux with GCC (although I suspect only when using _GLIBCXX_USE_CXX11_ABI=0). But this is definitely not the case with Clang and libc++ on Mac.

See comments in my [original DataView commit](https://github.com/GafferHQ/gaffer/commit/3352f33c5959166a84a959acd7e8d6054d73dbe0) and the [current implementation for StringData](https://github.com/ImageEngine/cortex/blob/master/src/IECoreImage/OpenImageIOAlgo.cpp#L226-L231).

Let's also add a test case to verify that this works - I don't think we expose it directly to Python, but we should be able to exercise this by writing a StringVectorData metadata item into an EXR file via ImageWriter and then reading it back via ImageReader. We should also get Travis set up to run tests on OSX - I can do that part though...

Reply:
So if I understand correctly,  I have to create storage for char* array or array access function to fetch the correct pointer?  I suppose I prefer the member or utility function to get a char* in the string array case.

Reply:
> I have to create storage for char* array?

Yes, this should be analogous to the existing `DataView::m_charPointer` member which is used for StringData, except you'll need `vector<char *> m_charPointers`. You'll fill that with pointers obtained the same way as in the [StringData case](https://github.com/ImageEngine/cortex/blob/master/src/IECoreImage/OpenImageIOAlgo.cpp#L226-L231), including obeying the `createUStrings` parameter - see [OSLExpressionEngine](https://github.com/GafferHQ/gaffer/commit/963bf1966f267ff8f0ee9a6edf05544f8c0eed65#diff-e979a46c72f9bc73ea0eb15965bc7b70) for the motivation for that.

> I suppose I prefer the member or utility function to get a char* in the string array case.

I don't know what you mean. It sounds like you're talking about not storing an array at all, but we need to give OIIO a pointer to an array of `char *`, so I don't know how that could work.

Reply:
> I don't know what you mean. It sounds like you're talking about not storing an array at all, but we need to give OIIO a pointer to an array of `char *`, so I don't know how that could work.

I think I must have only been thinking about the OSL / ShadingEngine uses rather than the wider picture. Thanks!

Reply:
> Yes, this should be analogous to the existing DataView::m_charPointer member which is used for StringData, except you'll need vector<char *> m_charPointers. You'll fill that with pointers obtained the same way as in the StringData case, including obeying the createUStrings parameter - see OSLExpressionEngine for the motivation for that.

Addressed in 6df87c5

Reply:
Can you add the test that I suggested please?

Reply:
Added test in 0ea49b3.